nameserver/nameserver.go

```diff
@@ -215,7 +217,7 @@ func (n *Nameserver) receiveGossip(msg []byte) (router.GossipData, router.Gossip
 		return nil, nil, fmt.Errorf("clock skew of %d detected", delta)
 	}
 
-	if err := gossip.Entries.check(); err != nil {
+	if err := check(CaseSensitive(gossip.Entries)); err != nil {
```

Comment:
We should drop this check. That way we switch to sending case-insensitive entries in 1.3, and get rid of most of of the code from this PR at that point.